src/Microsoft.Health.Dicom.Core/Features/Query/QueryParser.TagValueParser.cs

```diff
@@ -42,6 +43,64 @@ private static QueryFilterCondition ParseDateTagValue(QueryTag queryTag, string
             return new DateSingleValueMatchCondition(queryTag, parsedDate);
         }
 
+        private static QueryFilterCondition ParseDateTimeTagValue(QueryTag queryTag, string value)
+        {
```

Comment:
Can we consolidate these methods, they all look the same, except the parser is different

Reply:
This is doable but would make things a bit complicated for the fact that each of these methods has following differences:
- Parser
- Exception
- Single value match condition
- Range value match condition

I will proceed by combining Date and DateTime methods as they are fairly similar, however, will leave Time method as is as it uses long instead of DateTime which would make things messier.